src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/AwsConfigSection.cs

```diff
@@ -0,0 +1,25 @@
+namespace Kralizek.Extensions.Configuration.Internal
+{
+    /// <summary>
+    /// An object intended to bind an <see cref="IConfigurationSection"/> to.
+    /// </summary>
+    internal class AwsConfigSection
+    {
+        public const string DefaultConfigSectionName = "AWS";
+        
+        /// <summary>
+        /// The string representation of an AWS Region.
+        /// </summary>
+        public string Region { get; set; }
+        
+        /// <summary>
+        /// The name of the credentials profile to be used by the Secrets Manager client.
+        /// </summary>
+        public string Profile { get; set; }
+        
+        /// <summary>
+        /// The location of the file containing the credentials profiles.
+        /// </summary>
+        public string ProfilesLocation { get; set; }
+    }
```

Comment:
I've been thinking if it's better for the extension to handle everything or if it is worth taking a dependency on [AWSSDK.Extensions.NETCore.Setup](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-netcore.html) .

Unfortunately ```AWSOptions? options = settings.GetAWSOptions("AWS:SecretsManager");``` from the mentioned AWS package does not initialize `AWSCredentials` for some reason. 
To overcome this I added a new method to IConfiguration : `GetAWSOptionsWithCredentials` that also sets the options.Credentials.

Reply:
I'd like to keep this library as supple and lean as possible.